python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -0,0 +1,567 @@
+    TunableLaser, LEDProjector, LinearStage, Fiberspectrograph, Electrometer.
+
+    Parameters
+    ----------
+    domain : `lsst.ts.salobj.Domain`
+        Domain for remotes. If `None` create a domain.
+    log : `logging.Logger`
+        Optional logging class to be used for logging operations. If `None`,
+        creates a new logger. Useful to use in salobj.BaseScript and allow
+        logging in the class use the script logging.
+    intended_usage: `int`
+        Optional integer that maps to a list of intended operations. This is
+        used to limit the resources allocated by the class by gathering some
+        knowledge about the usage intention. By default allocates all
+        resources.
+    mtcamera: ComCam or LSSTCam
+    """
+
+    def __init__(
+        self,
+        domain: typing.Optional[salobj.Domain] = None,
+        log: typing.Optional[logging.Logger] = None,
+        intended_usage: typing.Optional[int] = None,
+        mtcamera: typing.Optional[ComCam] = None,
+    ) -> None:
+
+        self.electrometer_projector_index = 201
+        self.fiberspectrograph_blue_index = 1
+        self.fiberspectrograph_red_index = 2
+        self.linearstage_led_select_index = 1
+        self.linearstage_led_focus_index = 2
+        self.linearstage_laser_focus_index = 3
+        self.linearstage_select_index = 4
+
+        super().__init__(
+            components=[
+                "TunableLaser",
+                "LEDProjector",
+                f"FiberSpectrograph:{self.fiberspectrograph_blue_index}",
+                f"FiberSpectrograph:{self.fiberspectrograph_red_index}",
+                f"Electrometer:{self.electrometer_projector_index}",
+                f"LinearStage:{self.linearstage_led_select_index}",
+                f"LinearStage:{self.linearstage_led_focus_index}",
+                f"LinearStage:{self.linearstage_laser_focus_index}",
+                f"LinearStage:{self.linearstage_select_index}",
+            ],
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+        )
+
+        self.mtcamera = mtcamera
+        self.ls_select_led_location = 0.0  # um
+        self.ls_select_laser_location = 20.0  # um
+
+        self.laser_enclosure_temp = 20.0  # C
+
+        # ALl this needs to go in a config yaml file somewhere.
+        self.led_per_filter = {
+            "u": ["M385L3"],
+            "g": ["M455L4", "M505L4"],
+            "r": ["M565L3", "M660L4"],
+            "i": ["M730L5", "M780LP1"],
+            "z": ["M850L3", "M940L3"],
+            "y": ["M970L4"],
+        }
+        self.ls_led_locations = {
+            "u": 0.0,
+            "g": 1.0,
+            "r": 2.0,
+            "i": 3.0,
+            "z": 4.0,
+            "y": 5.0,
+        }
+        self.ls_led_focus = {"u": 0.0, "g": 1.0, "r": 2.0, "i": 3.0, "z": 4.0, "y": 5.0}
+
+        self.exptime_dict: dict[str, float] = dict(
```

Comment:
is this used anywhere?

Reply:
You have put the exposure times in the configuration file instead of this dictionary. I had originally thought the exposure times would be generated within the code.